### PR TITLE
Feature/getusers

### PR DESCRIPTION
**Pull request by RvdK**

### CODE REVIEW

File: BankChain/.idea/checkstyle-idea.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (MatthijsKok) [inline BankChain/.idea/checkstyle-idea.xml]: please remove this file. We already have a config in /config/quality/checkstyle

File: BankChain/app/src/androidTest/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqTest.java [hunk 2]
```diff
@@ -126,4 +126,6 @@ public void testSession() throws Exception {
     }
 
 
+
```
Comment (joskuijpers) [inline BankChain/app/src/androidTest/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqTest.java]: This seems useless

Comment (MatthijsKok) [inline BankChain/app/src/androidTest/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqTest.java]: Why is this here? please remove

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/MainActivity.java [hunk 3]
```diff
@@ -42,13 +46,20 @@ protected void onCreate(Bundle savedInstanceState) {
                     Toast.makeText(getApplicationContext(),
                             "Created session!",
                             Toast.LENGTH_LONG).show();
-
+                    /*
```
Comment (MatthijsKok) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/MainActivity.java]: Please don't commit commented code. Remove completely. We always have this code in our history if needed.

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqAccount.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ */
+
+public class BunqAccount implements Account{
+
+    private Party party;
+    private String Iban;
+
+    @Override
+    public String getIban() {
+        return null;
```
Comment (MatthijsKok) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqAccount.java]: why does this return null?

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqInterceptor.java [hunk 5]
```diff
@@ -156,7 +156,7 @@ private String createSignable(String prefix, Headers headers,
         calculating them.
 
         Cache-Control
-        User-Agent
+        Party-Agent
```
Comment (MatthijsKok) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqInterceptor.java]: Party-Agent is not a valid HTML header i believe.

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqParty.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+
+    public BunqParty(UserService.ListResponse.UserCompany user){
+        id = user.id;
+        name = user.name;
+    }
+
+    @Override
+    public String getName() {
+        return null;
```
Comment (MatthijsKok) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqParty.java]: Why do these implementations have no actual implementation stubs can be generated easily, no need to commit them beforehand

**Review by MatthijsKok — DISMISSED** [review]
Small questions and comments

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/Account.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package nl.tudelft.ewi.ds.bankchain.bank;
+
+/**
+ * Created by Richard-HP on 15/05/2017.
+ */
+
+public interface Account {
```
Comment (joskuijpers) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/Account.java]: No documentation

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/Bank.java [hunk 8]
```diff
@@ -20,6 +20,8 @@
     public abstract CompletableFuture<? extends Session> createSession();
 
     public abstract CompletableFuture<List<? extends Transaction>> listTransactions();
+
+    public abstract CompletableFuture<List<? extends Party>>   listUsers();
```
Comment (joskuijpers) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/Bank.java]: Space too much

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/Party.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+     * @return ID or 0
+     */
+    int getId();
+
+    /**
+     * if the user is linked to this api key return the list of all connected accounts
+     * @return
+     */
+    List<Account> getAccounts();
+
```
Comment (joskuijpers) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/Party.java]: Too many newlines

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/Party.java [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.List;
+
+/**
+ * Created by Richard-HP on 15/05/2017.
+ */
+
+public interface Party {
+
+
```
Comment (joskuijpers) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/Party.java]: Too many newlines

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqAccount.java [hunk 11]
```diff
@@ -0,0 +1,23 @@
+package nl.tudelft.ewi.ds.bankchain.bank.bunq;
+
+import nl.tudelft.ewi.ds.bankchain.bank.Account;
+import nl.tudelft.ewi.ds.bankchain.bank.Party;
+
+/**
+ * Created by Richard-HP on 15/05/2017.
+ */
+
+public class BunqAccount implements Account{
```
Comment (joskuijpers) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqAccount.java]: No documentation. Space needed after Account. (Run checkstyle from the project)

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqAccount.java (hunk 4 above)
Comment (RvdK) [reply BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqAccount.java]: this is not yet implemented correctly will be done in the get account sotry

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqInterceptor.java (hunk 5 above)
Comment (joskuijpers) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqInterceptor.java]: Wat?

Comment (RvdK) [reply BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqInterceptor.java]: changed

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqParty.java (hunk 6 above)
Comment (RvdK) [reply BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqParty.java]: you are right have changed it now

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/MainActivity.java (hunk 3 above)
Comment (RvdK) [reply BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/MainActivity.java]: I still have to do the merge and this part ha been completely refactored so it will be removed at that point

**Comment by MatthijsKok** [comment]
After these changes can you try to rebase onto develop? travis should work in develop now

**Review by joskuijpers — APPROVED** [review]
